source/Irrlicht/CMakeLists.txt

```diff
@@ -457,6 +456,10 @@ add_library(IRROTHEROBJ OBJECT
 	os.cpp
 )
 
+if(LINUX_PLATFORM)
```

Comment:
Maybe check the `DEVICE` variable instead?

Reply:
i can, but as we just setuped it for purposes like this i thought we can :)

Reply:
if you look code, ti's enabled with EMSCRIPTEN or in default case for the device (L98 and L105)

Reply:
> if you look code

if you look at `git blame`...

> i can, but as we just setuped it for purposes like this i thought we can :)

don’t understand what you mean, sorry.

What I mean is, this `if(LINUX_PLATFORM)` duplicates a part of the device selection logic (which is at lines 83–117 currently).

Reply:
Also, `CIrrDeviceLinux` is a misnomer. It’s an X11 device. It won’t work on Emscripten. (not that Emscripten is officially supported but...)

Reply:
okay i'll study that part